### PR TITLE
Fix: Remove SQS policy if empty attribute is passed

### LINKED ISSUE #4225: Error: error waiting for SQS Queue Policy to delete: Error unmarshaling policy: unexpected end of JSON input
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description

I am using :
* LocalStack: 
```
localstack                        0.12.13.1
localstack-client              1.20
localstack-ext                  0.12.10.11
```
* Terraform: 
```
Terraform v1.0.1
on darwin_amd64
+ provider registry.terraform.io/hashicorp/aws v3.47.0
```

When setting up an infrastructure containing only a s3 bucket with an associated event notification mechanism to a sqs queue, the destruction of this infrastructure fails using LocalStack (works fine using AWS). The resulting error message is

```
aws_sqs_queue_policy.distributor: Still destroying... [id=http://localhost:4566/000000000000/distributor_queue, 10s elapsed]
aws_sqs_queue_policy.distributor: Still destroying... [id=http://localhost:4566/000000000000/distributor_queue, 20s elapsed]
aws_sqs_queue_policy.distributor: Still destroying... [id=http://localhost:4566/000000000000/distributor_queue, 30s elapsed]
aws_sqs_queue_policy.distributor: Still destroying... [id=http://localhost:4566/000000000000/distributor_queue, 40s elapsed]
aws_sqs_queue_policy.distributor: Still destroying... [id=http://localhost:4566/000000000000/distributor_queue, 50s elapsed]
aws_sqs_queue_policy.distributor: Still destroying... [id=http://localhost:4566/000000000000/distributor_queue, 1m0s elapsed]
╷
│ Error: error waiting for SQS Queue Policy (http://localhost:4566/000000000000/distributor_queue) to delete: Error unmarshaling policy: unexpected end of JSON input
│ 
```

## Expected behavior

I would expect the destruction of the infrastructure to go through without any errors (as is the case when deploying using AWS).

## Actual behavior

Getting an error message related to the aws_sqs_queue_policy `distributor`: 
```
│ Error: error waiting for SQS Queue Policy (http://localhost:4566/000000000000/distributor_queue) to delete: Error unmarshaling policy: unexpected end of JSON input
```

# Steps to reproduce

## Command used to start LocalStack

```
export SERVICES=sqs,s3,iam,kinesis,sts && \
localstack start
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

The following are the terraform files used in the infrastructure. Put them in a directory and run `terraform init && terraform apply` to construct the infrastructure and later on call `terraform destroy` to destroy the infrastructure and witness the error message.

aws.tf
```
provider "aws" {
  access_key                  = "mock_access_key"
  region                      = "eu-central-1"
  s3_force_path_style         = true
  secret_key                  = "mock_secret_key"
  skip_credentials_validation = true
  skip_metadata_api_check     = true
  skip_requesting_account_id  = true

  endpoints {
    apigateway     = "http://localhost:4566"
    cloudwatch     = "http://localhost:4566"
    es             = "http://localhost:4566"
    firehose       = "http://localhost:4566"
    iam            = "http://localhost:4566"
    kinesis        = "http://localhost:4566"
    s3             = "http://localhost:4566"
    secretsmanager = "http://localhost:4566"
    sns            = "http://localhost:4566"
    sqs            = "http://localhost:4566"
    ssm            = "http://localhost:4566"
    stepfunctions  = "http://localhost:4566"
    sts            = "http://localhost:4566"
  }
}
```

s3.tf
```
resource "aws_s3_bucket" "local_test_events" {
  bucket        = "local-test-events"
  acl = "private"
  force_destroy = true        
} 
  
  
resource "aws_s3_bucket_public_access_block" "s3_privacy" {
  bucket = aws_s3_bucket.local_test_events.id
    
  block_public_acls   = true
  block_public_policy = true
    
  depends_on = [   
    aws_s3_bucket.local_test_events
  ] 
}   
    
resource "aws_s3_bucket_notification" "event_dump" {
  bucket = aws_s3_bucket.local_test_events.id
    
  queue {          
    queue_arn     = aws_sqs_queue.distributor.arn
    events        = ["s3:ObjectCreated:*"]
  }

  depends_on = [
            aws_s3_bucket.local_test_events,
            aws_s3_bucket_public_access_block.s3_privacy,
            aws_sqs_queue.distributor,
            aws_sqs_queue_policy.distributor
  ]
}
```

sqs.tf
```
resource "aws_sqs_queue" "distributor" {
  name = "distributor_queue"
} 
  

resource "aws_sqs_queue_policy" "distributor" {
  queue_url = aws_sqs_queue.distributor.id
  policy    = data.aws_iam_policy_document.sqs_distributor.json
} 
    
data "aws_iam_policy_document" "sqs_distributor" {
  policy_id = "__default_policy_ID"
  statement {
    actions = [
      "sqs:SendMessage",
    ]
    condition {
      test     = "ArnEquals"
      variable = "aws:SourceArn"
  
      values = [
        "${aws_s3_bucket.local_test_events.arn}",
      ]
    }
    effect = "Allow"
    principals {
      type = "AWS"
      identifiers = [
      "*"]  
    }       
    resources = [
      "${aws_sqs_queue.distributor.arn}",
    ]
    sid = "__default_statement_ID"
  }
}
```

Thanks a lot for your help and let me know if I can be of any assistance to solve the issue!

### PR DESCRIPTION
**Pull request by mgagliardo**
Fixes #4225 .

Description:
There is a problem when trying to remove an SQS queue if it has policy attached, and it is basically because terraform tries to set `Policy=None` before actually removing the queue.

[This explanation](https://github.com/localstack/localstack/issues/4225#issuecomment-871415857) states pretty well the issue coming from [this piece of terraform code](https://github.com/hashicorp/terraform-provider-aws/blob/ccc13f55709c860a2130ecee6e21acde2318e936/aws/resource_aws_sqs_queue_policy.go#L104)

Main issue for us is the `_set_attributes` function does not detect the attribute being passed an empty K/V if passing i.e. `Policy=""` or `DelaySeconds=""` which is the behaviour by design on the AWS CLI.

**Pros:**
Actually removing the policy if we pass an empty attribute.

**Cons:**
`Any` empty attribute will actually remove the policy.
Example:
```
awslocal sqs set-queue-attributes --queue-url http://localhost:4566/000000000000/myqueue  --attributes DelaySeconds=""
```

One side note is that no should pass empty parameters, but if it would be the case (by mistake for example) it would remove their policy.

Suggestions are welcome

### CODE REVIEW

File: localstack/services/sqs/sqs_starter.py [hunk 1]
```diff
@@ -65,6 +65,9 @@ def add_message(self, *args, **kwargs):
     def _set_attributes(self, attributes, now=None):
         _set_attributes_orig(self, attributes, now)
 
+        if attributes == {}:
```
Comment (whummer) [inline localstack/services/sqs/sqs_starter.py:68]: Thanks for digging into this, @mgagliardo ! 👍 

Not entirely sure if this is the right approach here. Looking at the definition of `sqsQueueEmptyPolicyAttributes` [here](https://github.com/hashicorp/terraform-provider-aws/blob/ccc13f55709c860a2130ecee6e21acde2318e936/aws/resource_aws_sqs_queue_policy.go#L17-L21) and how it is being used in that file, I think we should rather check if `attributes` contains a `Policy` attribute with an empty string:

```
if attributes.get('Policy') == '':
    self.policy = None
```

Can you please double-check if this change fixes the issue with Terraform? Thanks!

File: tests/integration/test_sqs.py [hunk 2]
```diff
@@ -233,6 +233,12 @@ def test_set_queue_policy(self):
         attrs = self.client.get_queue_attributes(QueueUrl=queue_url, AttributeNames=['Policy'])['Attributes']
         self.assertIn('sqs:SendMessage', attrs['Policy'])
 
+        # Passing empty attributes resets the to defaults attributes with no Policy
+        empty_attributes = {}
```
Comment (whummer) [inline tests/integration/test_sqs.py:237]: This should be changed to:
```
        # Passing a Policy with empty string resets the policy of the queue
        reset_policy_attributes = {
            'Policy': ''
        }
```

**Review by whummer — CHANGES_REQUESTED** [review]

File: localstack/services/sqs/sqs_starter.py (hunk 1 above)
Comment (mgagliardo) [reply localstack/services/sqs/sqs_starter.py:68]: I have already checked that and sadly when you send an empty attribute (i.e. `--attributes Policy=""` or even trying a terraform destroy to trigger this) :

Just a simple log, when creating the policy (setting attributes):
```
Attributes passed: {'ContentBasedDeduplication': 'false', 'DelaySeconds': '0', 'FifoQueue': 'false', 'KmsDataKeyReusePeriodSeconds': 300, 'KmsMasterKeyId': None, 'MaximumMessageSize': '262144', 'MessageRetentionPeriod': '345600', 'Policy': None, 'ReceiveMessageWaitTimeSeconds': '0', 'RedrivePolicy': None, 'VisibilityTimeout': '30'}

Attributes passed: {'Policy': '{\n  "Version": "2012-10-17",\n  "Id": "__default_policy_ID",\n  "Statement": [\n    {\n      "Sid": "__default_statement_ID",\n      "Effect": "Allow",\n      "Action": "sqs:SendMessage",\n      "Resource": "arn:aws:sqs:us-east-1:000000000000:distributor-queue",\n      "Principal": {\n        "AWS": "*"\n      },\n      "Condition": {\n        "ArnEquals": {\n          "aws:SourceArn": [\n            "arn:aws:s3:::local-test-events",\n            "arn:aws:s3:potato"\n          ]\n        }\n      }\n    }\n  ]\n}'}
```

And then when destroying:
```
Attributes passed: {}
```

**Comment by thrau** [comment]
we've reformatted the codebase with black. if you need help rebasing your branch, please let me know. maybe this post is helpful: https://github.com/localstack/localstack/pull/4252#issuecomment-873175489

**Pull request closed without merging**